lib/cucumber/formatter/backtrace_filter.rb

```diff
@@ -22,6 +22,7 @@ module Formatter
     @backtrace_filters << RbConfig::CONFIG['rubylibdir'] if RbConfig::CONFIG['rubylibdir']
 
     @backtrace_filters << 'org/jruby/' if ::Cucumber::JRUBY
+    @backtrace_filters << '<internal:core>' if RUBY_ENGINE == "truffleruby"
```

Comment:
@eregon do you think it's safe change? It does the job (all internal errors are filtered out and tests are passing), but later on I noticed that  some TypeError may dissapear as well, for example :
```
<internal:core> core/marshal.rb:51:in `__marshal__': singleton class can't be dumped (TypeError)
	from <internal:core> core/marshal.rb:1006:in `serialize'
	from <internal:core> core/marshal.rb:1309:in `dump'
	from -e:1:in `<main>'
```

Reply:
Could you please fix the rubocop error (use single quotes), and add truffleruby to the [GitHub Actions build matrix](https://github.com/cucumber/cucumber-ruby/blob/main/.github/workflows/cucumber-ruby.yml)?

Reply:
done

Reply:
It should be any path starting with `<internal:`, that's also used by CRuby:
```
$ ruby -ve 'p method(:then).source_location'
ruby 3.0.2p107 (2021-07-07 revision 0db68f0233) [x86_64-linux]
["<internal:kernel>", 120]
```

Since it's converted to a Regexp later, then `'^<internal:'` is best.